message_ix_models/tests/model/water/data/test_water_data.py

```diff
@@ -0,0 +1,37 @@
+import os
+
+from message_ix_models.util import package_data_path
+
+
+def check_files_in_folder(folder_path):
```

Comment:
```suggestion
def check_files_in_folder(folder_path: Path) -> None:
    """Check that every `_R11` file has a `_R12` counterpart."""
```
Always good to provide type hints, which allow automatic tools to catch errors and others to understand your code. 
Also always good to provide docstrings to help other understand your code.
Finally, you could consider renaming that function since it doesn't actually check if *general* files are in `folder_path`, but specifically compares `R11` and `R12` files. So maybe `check_R12_files_complete`?